### PR TITLE
fix versioned buckets: list, rename, delete

### PR DESCRIPTION
**Pull request by Zhell1**
- make [list, rename, delete] actions on jupyterlab work with versioned buckets (and not display delete markers)
- add minio-distributed environement in makefile to make testing versioning possible
- add some logging & doc for the pain points I encountered

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -40,6 +51,19 @@ Start Jupyter Notebook in another terminal:
...
+
+```
+conda activate s3contents && make env && jupyter notebook
+```
+
+or use this faster command to copy changes to s3contents installation path:
+
+```bash
+rsync -r --exclude '.git' ./s3contents/ $(echo "$(pip show s3contents | grep Location: | cut -d' ' -f2)/s3contents/") && jupyter notebook
+
```
Comment (danielfrg) [inline CONTRIBUTING.md:64]: Why is this part needed? If you do `make env` poetry will install the package in edit mode so you should not need this.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,11 +2,15 @@
...
 ## Development environment
 
+The following requires poetry 1.2.0a2 or newer.
+
 Create Python env
 
 ```shell
 make env
 source ./.venv/bin/activate
+conda create --name s3contents
```
Comment (danielfrg) [inline CONTRIBUTING.md:12]: `make env` uses poetry now instead of conda so i dont think this is needed.

**Review by danielfrg — APPROVED** [review]

**Comment by danielfrg** [comment]
Thank you very much for the PR. I made a couple of small comments on docs and linting but nothing major.

Since this uses MinIO for testing it would be cool to have CI for it, probably in this case it would mean a different github action. It would be interesting but not necessary.

File: Makefile [hunk 3]
```diff
@@ -66,8 +66,32 @@ reset: clean  ## Reset Python
...
+		"/data2" \
+		"/data3" \
+		"/data4" \
+		--console-address ":9001"
 
 
 help:  ## Show this help menu
-	@grep -E '^[0-9a-zA-Z_-]+:.*?##.*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?##"; OFS="\t\t"}; {printf "\033[36m%-30s\033[0m %s\n", $$1, ($$2==""?"":$$2)}'
+	@grep -E '^[0-9a-zA-Z_-]+:.*?##.*$$' $(MAKEFILE_LIST) | sort \
+		| awk 'BEGIN {FS = ":.*?##"; OFS="\t\t"}; {printf "\033[36m%-30s\033[0m %s\n", $$1, ($$2==""?"":$$2)}'
```
[the remark this answers is not in the document]
Comment (Zhell1) [reply Makefile]: done 👍

File: CONTRIBUTING.md (hunk 2 above)
Comment (Zhell1) [reply CONTRIBUTING.md:17]: I have unable to make it work, after `make env` I don't have `./.venv/bin/activate` file, which makes sourcing impossible

I separated the explanation to make it clear that this is not the preferred solution but still keep it for people who are in my case for now. Is that ok ?

File: CONTRIBUTING.md (hunk 1 above)
Comment (Zhell1) [reply CONTRIBUTING.md:73]: see below comment, separated this and explained that this is a fallback method.

File: CONTRIBUTING.md (hunk 2 above)
Comment (danielfrg) [reply CONTRIBUTING.md:17]: Ah got it. It makes sense because i have some global poetry config.
This is perfect, thank you!

**Comment by danielfrg** [comment]
I fixed the CI issues so I am just going to merge since tests passed before you made the doc fixes.

Thank you for the PR!

**Comment by Zhell1** [comment]
Thank you for such a quick reaction on this, you made contributing an awesome experience 🙏

**Comment by danielfrg** [comment]
Thanks! I will do a release hopefully tomorrow.

**Comment by danielfrg** [comment]
Version 0.9.0 is released including this.
